lib/pdf/collate.rb

```diff
@@ -44,13 +47,13 @@ def self.merge_pdfs(output_file, *filenames)
     temp_file_handles = []
 
     filenames.map! do |filename|
-      next filename unless URI.parse(filename).scheme == 'http'
+      next filename unless string_is_url(filename)
       begin
         local_file = Tempfile.from_url(filename)
       rescue Exception => msg
         puts "Error downloading PDF file #{filename} for output file #{output_file}. Aborting"
         puts "Error message: #{msg}"
-        exit 1
+        raise
```

Comment:
Why is `raise` better than `exit 1` here? Seems like each would cause the program to exit with failure status, unless you're re-rescuing this somewhere, and you're already printing out the stack trace in the lines above.